src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1506 @@
+          local return_string = finale.FCString()
+          return_string.LuaString = harp_notes:GetText(return_string)
+          strings_read()
+          if partial_checkbox:GetCheck() == 1 then partial = true
+          elseif partial_checkbox:GetCheck() == 0 then partial = false end
+          if diagram_checkbox:GetCheck() == 1 then diag = true
+          else diag = false end
+          local root = root_calc()
+          local scaleinfo = ""
+          if return_string.LuaString ~= "" then
+            direct = true
+            process_return(return_string.LuaString)
+            strings_read()
+          else
+            if scale_check:GetCheck() == 1 then 
+              scaleinfo = root.LuaString.." "..scales[sel_scale:GetSelectedItem() + 1]
+            elseif chord_check:GetCheck() == 1 then 
+              scaleinfo = root.LuaString..chords[sel_chord:GetSelectedItem() + 1]
+            end
+          end
+
+          harp_diagram(str.LuaString, diag, scaleinfo, partial)
+
+          if partial and not changes then
+            if not override then
+              ui:AlertInfo("There are no pedal changes required. Try entering notes directly in the 'Enter Notes' field, or update your 'last used' pedals somewhere else and try again.", nil)
+            end
+            goto error
+          end
+          pedals_add(diag, partial)
+          ::error::
+          str.LuaString = ""
+          harp_notes:SetText(str)
+          changes_static:SetText(str)
+          for i = 1, 6, 1 do
+            str.LuaString = str.LuaString..harpstrings[i]..", "
+          end
+          str.LuaString = str.LuaString..harpstrings[7]
+          config.last_notes = str.LuaString
+          configuration.save_user_settings(script_name, config)
+          finenv.Region():Redraw()
+          direct = false
+          override = false
+          changes = false
+          direct_notes = {0, 0, 0, 0, 0, 0, 0}
+        end -- apply()
+
+        dialog:RegisterHandleCommand(callback)
+        dialog:RegisterHandleOkButtonPressed(callback_ok)
+        dialog:RegisterHandleDataListSelect(callback_update)
+        if dialog.RegisterCloseWindow then
+          dialog:RegisterCloseWindow(on_close)
+        end
+        -- initialize pedals
+        partial = false 
+        harp_diagram(config.last_notes, diag, scaleinfo, partial)
+        update_variables()
+
+        if partial_checkbox:GetCheck() == 1 then partial = true end
+        pedals_update()
+
+        dialog.OkButtonCanClose = false
+        finenv.RegisterModelessDialog(dialog) -- must register, or ShowModeless does nothing
+        dialog:ShowModeless()
+        if nil ~= finenv.RetainLuaState then -- if finenv.RetainLuaState then --same?
+          finenv.RetainLuaState = true
+        end
+
+--                if dialog:ExecuteModal(nil) == finale.EXECMODAL_OK then
+--                end -- if dialog:Execute...
+
+      end -- get_harp
+      harp_dialog()
+    end -- harp
+
+    harp()
```

Comment:
Seems like there might be some "end" statements missing, or a few "unindents"? It seems weird to have a file end with lines still indented.

Reply:
What are unindents?

All I know is that if I add any more 'end' statements it breaks...